src/extension.ts

```diff
@@ -308,4 +315,46 @@ export async function activate(ctx: vscode.ExtensionContext) {
 
     explorer.refresh();
   }
+
+  async function pickExecutable() {
```

Comment:
This function is much better than pickTestOrBenchmark() above, would be nice if we could have some common code. After all they both select a task, it's just the mode that change.

Reply:
I think we can do a massive simplification: https://github.com/mesonbuild/vscode-meson/commit/13a9fd0433062c5be2afaa776be1113c913104b0

Reply:
I refactored that in https://github.com/mesonbuild/vscode-meson/pull/127, I don't think it is a blocker to get this PR merged first.